src/connector/src/parser/scalar_adapter.rs

```diff
@@ -104,134 +112,185 @@ impl ToSql for ScalarAdapter<'_> {
+    pub fn into_scalar(self, ty: &DataType) -> Option<ScalarImpl> {
+        match (self, &ty) {
+            (ScalarAdapter::Builtin(scalar), _) => Some(scalar),
             (ScalarAdapter::Uuid(uuid), &DataType::Varchar) => {
                 Some(ScalarImpl::from(uuid.to_string()))
             }
             (ScalarAdapter::Numeric(numeric), &DataType::Varchar) => {
-                Some(ScalarImpl::from(pg_numeric_to_string(numeric)))
+                Some(ScalarImpl::from(pg_numeric_to_string(&numeric)))
             }
             (ScalarAdapter::Numeric(numeric), &DataType::Int256) => {
-                pg_numeric_to_rw_int256(numeric)
+                pg_numeric_to_rw_int256(&numeric)
+            }
+            (ScalarAdapter::Numeric(numeric), &DataType::Decimal) => {
+                pg_numeric_to_rw_numeric(&numeric)
             }
             (ScalarAdapter::Enum(EnumString(s)), &DataType::Varchar) => Some(ScalarImpl::from(s)),
-            _ => {
+            (ScalarAdapter::NumericList(vec), &DataType::List(dtype)) => {
+                let mut builder = dtype.create_array_builder(0);
+                for val in vec {
+                    let scalar = match (val, &dtype) {
+                        (Some(numeric), box DataType::Varchar | box DataType::Int256 | box DataType::Decimal) => {
+                            if pg_numeric_is_special(&numeric) {
+                                return None;
+                            } else {
+                                ScalarAdapter::Numeric(numeric).into_scalar(&dtype)
+                            }
+                        }
+                        (Some(_), _) => unreachable!(
+                            "into_scalar_in_list should only be called with ScalarAdapter::Numeric types"
+                        ),
+                        (None, _) => None,
+                    };
+                    builder.append(scalar);
+                }
+                Some(ScalarImpl::from(ListValue::new(builder.finish())))
+            }
+            (ScalarAdapter::EnumList(vec), &DataType::List(dtype)) => {
+                let mut builder = dtype.create_array_builder(0);
+                for val in vec {
+                    match val {
+                        Some(EnumString(s)) => {
+                            builder.append(Some(ScalarImpl::from(s)));
+                        }
+                        None => {
+                            return None;
+                        }
+                    }
+                }
+                Some(ScalarImpl::from(ListValue::new(builder.finish())))
+            }
+            (ScalarAdapter::List(vec), &DataType::List(dtype)) => {
+                // Due to https://github.com/risingwavelabs/risingwave/issues/16882, INTERVAL_ARRAY is not supported in Debezium, so we keep backfilling and CDC consistent.
+                if matches!(**dtype, DataType::Interval) {
```

Comment:
I think the `into_scalar` just need to convert the value of `ScalarAdapter` to `ScalarImpl`, and don't need to align the behavior of Debezium. Because we may need to sink a `Interval` list to downstream. Would it cause problem if we don't `return None` here?

Reply:
There would cause data inconsistency. Two rows with the same data converted to different rows. If the first row is `insert` and the second is `delete`, may cause nagative sum count in agg operator.